src/frontend/src/pages/GanttPage/Gantt.tsx

```diff
@@ -0,0 +1,229 @@
+import { addDays, differenceInDays, eachDayOfInterval, isMonday } from 'date-fns';
+import { ComponentProps, DragEvent, MouseEvent, useEffect, useState } from 'react';
+import useMeasure from 'react-use-measure';
+import {
+  GANTT_CHART_GAP_SIZE,
+  GANTT_CHART_CELL_SIZE,
+  applyChangesToEvents,
+  EventChange,
+  Task
+} from '../../utils/gantt.utils';
+import useId from '@mui/material/utils/useId';
+import { Box, Typography, useTheme } from '@mui/material';
+import { grey } from '@mui/material/colors';
+import { dateToString } from '../../utils/datetime.utils';
+
+interface GanttProps {
+  start: Date;
+  end: Date;
+  tasks: Task[];
+  isEditMode: boolean;
+  saveChanges: (eventChanges: EventChange[]) => void;
+}
+
+export function Gantt({ start, end, tasks, isEditMode, saveChanges }: GanttProps) {
+  const days = eachDayOfInterval({ start, end }).filter((day) => isMonday(day));
+  const [eventChanges, setEventChanges] = useState<EventChange[]>([]);
+  const createChange = (change: EventChange) => {
+    setEventChanges([...eventChanges, change]);
+  };
+
+  useEffect(() => {
+    // only try to save changes when we're going from non-editing to editing mode
+    if (!isEditMode) {
+      saveChanges(eventChanges);
+      setEventChanges([]); // reset the changes after sending them
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isEditMode]);
+
+  const displayEvents = applyChangesToEvents(tasks, eventChanges);
+
+  return (
+    <Box sx={{ width: 'fit-content' }}>
+      {/* Data display: reset list of events every time eventChanges list changes using key */}
+      <Box sx={{ mt: '1rem', width: 'fit-content' }} key={eventChanges.length}>
+        {displayEvents.map((event) => {
+          return <Event key={event.id} days={days} event={event} isEditMode={isEditMode} createChange={createChange} />;
+        })}
+      </Box>
+    </Box>
+  );
+}
+
+function Event({
+  days,
+  event,
+  createChange,
+  isEditMode,
+  ...props
+}: { days: Date[]; event: Task; createChange: (change: EventChange) => void; isEditMode: boolean } & ComponentProps<'div'>) {
```

Comment:
ok next steps are naming and organizing this code a lot better, so people can do work in it. It looks like this file is for one team section on the Gantt Chart (correct me if I'm wrong). This event component should be moved into another file and called GanttChartElement or GanttChartActivity or something (it should also be changed to be an arrow method